pkg/util/labels.go

```diff
@@ -0,0 +1,12 @@
+package util
+
+import metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+func DeepCopyLabels(src, tgt *metav1.ObjectMeta) {
+	if src.Labels != nil {
+		tgt.Labels = make(map[string]string, len(src.Labels))
```

Comment:
It would be safer to "append" to the target labels instead of overwriting - i.e. check if already exists before creating...

Reply:
We are checking the deep equality for labels since we are so opinionated about the labels of our operands. This was a result of a discussion with @tiraboschi

Reply:
we talked this with @zcahana  and we are ok with it.